### ISSUE TITLE
Feature Request: Site Blocklist

### ISSUE THREAD

**Issue by Ryonez** [ajayyy/SponsorBlock#166]
The ability to blacklist the add-on from function on some sites.

Use case: www.watchtogether.com

A site that syncs youtube video timelines between users. SponserBlock interferes with it and only recognizes the first video (If the first video has sponsor time, those time are skipped for all videos, even newly loaded ones). Being able to tell SponserBlock not to run on those sites would be required fro it to function without issues for everyone.

**Comment by ajayyy** [comment]
Would this still be required if support for those type of embeds is complete?

**Comment by Ryonez** [comment]
It help, but I think the blacklist would still be preferable. The site in the example controls plackback for several people, interference with it has proven to be slightly frustrating for others.

**Comment by mirror176** [comment]
I presume the mentioned site should have been watch2gether.com? I'm not familiar with it so not sure. watchtogether.com has a godaddy squat page.

**Comment by Ryonez** [comment]
@mirror176 You are correct, thank you for letting me know. I'll correct the post description.

**Comment by ajayyy** [comment]
Now that I think about it, this may not be possible. The extension doesn't have permission to view the URLs of the sites you are visiting so it cannot really determine things like this.

**Comment by ajayyy** [comment]
Maybe https://developer.chrome.com/apps/permissions can be used?

**Comment by OfficialNoob** [comment]
Could `document.referrer` be used?

**Comment by Bluscream** [comment]
I want this feature too. People are complaining that i'm skipping their beloved sponsorships ._.

**Comment by SomeNameDee** [comment]
#### Problem
Site-by-site blocklisting used to be a problem because the browser sees the YouTube site and YouTube embeds as both originating from YouTube.
#### New Development
Embeds have a property called "`window.location.ancestorOrigins`" (see [here](https://developer.mozilla.org/en-US/docs/Web/API/Location/ancestorOrigins)), which reports the parent site that the embed is contained within.
This was not a viable solution when this issue was first opened, as Edge and Firefox did not include this property.
Edge added it in 2020, and, as of Feb 24, 2026, Firefox became the last major browser to support the property, meaning it is now reasonable to use for this purpose.
Some details of this idea were discussed [here](https://discord.com/channels/603643120093233162/603643180663177220/1522945892745744454) on the SponsorBlock Discord.
#### Implementation
The property could either be given a dedicated whitelist/blacklist option in SponsorBlock's Behavior menu, or it could be added to the list of supported attributes for the Advanced Skip options, or possibly both.
